volatility/framework/plugins/windows/bigpools.py

```diff
@@ -0,0 +1,133 @@
+
+from volatility.framework import interfaces, renderers, exceptions, symbols
+from volatility.framework.configuration import requirements
+from volatility.framework.renderers import format_hints
+from volatility.framework.interfaces import configuration
+from volatility.framework.symbols import intermed
+from volatility.framework.symbols.windows import extensions
+from volatility.plugins.windows import poolscanner
+
+vollog = logging.getLogger(__name__)
+
+
+class BigPools(interfaces.plugins.PluginInterface):
+    """List big page pools."""
+
+    _version = (1, 0, 0)
+
+    is_vista_or_later = poolscanner.os_distinguisher(version_check = lambda x: x >= (6, 0),
+                                                     fallback_checks = [("KdCopyDataBlock", None, True)])
+
+    is_win10 = poolscanner.os_distinguisher(version_check = lambda x: (10, 0) <= x,
+                                            fallback_checks = [("ObHeaderCookie", None, True),
+                                                               ("_HANDLE_TABLE", "HandleCount", False)])
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        # Since we're calling the plugin, make sure we have the plugin's requirements
+        return [
+            requirements.TranslationLayerRequirement(name = 'primary',
+                                                     description = 'Memory layer for the kernel',
+                                                     architectures = ["Intel32", "Intel64"]),
+            requirements.SymbolTableRequirement(name = "nt_symbols", description = "Windows kernel symbols"),
+            requirements.StringRequirement(name = 'tags',
+                                           description = "Comma separated list of pool tags to filter pools returned",
+                                           optional = True,
+                                           default = None)
+        ]
+
+    @classmethod
+    def list_big_pools(cls,
+                       context: interfaces.context.ContextInterface,
+                       layer_name: str,
+                       symbol_table: str,
+                       tags: Optional[list] = None):
+        """Returns the big page pool objects from the kernel PoolBigPageTable array.
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            layer_name: The name of the layer on which to operate
+            symbol_table: The name of the table containing the kernel symbols
+            tags: An optional list of pool tags to filter big page pool tags by
+
+        Yields:
+            A big page pool object
+        """
+        kvo = context.layers[layer_name].config['kernel_virtual_offset']
+        ntkrnlmp = context.module(symbol_table, layer_name = layer_name, offset = kvo)
+
+        big_page_table_offset = ntkrnlmp.get_symbol("PoolBigPageTable").address
```

Comment:
Can we rely on this symbol always existing?  Not something to worry about here particularly, but I can't tell if we need error handling around it...